### DISCUSSION TITLE
Multi deploy

### DISCUSSION THREAD

**Discussion by SiMoSiMo2** [deployphp/deployer#2600] [General]
My needs:
I have one repo with 2 main folders: frontend and backend. On the backend there is a symfony application, while on the frontend there is only HTML and jQuery (for now). So I need to have 2 separate deploy tasks because I need to run some symfony tasks for the backend and some other for the frontend when I want to deploy the code.

I took some ideas from #1383, but I have to adapt them to the new version (7) because what I have coded is not producing what I want.

This is a piece of my `deploy.php`:
```
...

host('frontend')
    ->setHostname('***')
    ->set('deploy_path', '~/frontend')
    ->set('roles', 'front')
    ->set('stage', 'prod')
    ->setRemoteUser('***')
    ->setIdentityFile('***');

host('backend')
    ->setHostname('***')
    ->set('deploy_path', '~/backend')
    ->set('roles', 'back')
    ->set('stage', 'prod')
    ->setRemoteUser('***')
    ->setIdentityFile('***');

// Tasks
task('deploy:frontend', [
        'deploy:info',
        'deploy:prepare',
        'deploy:publish',
    ])->select('stage=prod, roles=front')
    ->desc('Deploy frontend project')
;

task('deploy:backend', [
        'deploy:info',
        'deploy:prepare',
        'deploy:publish'
    ])->select('stage=prod, roles=back')
    ->desc('Deploy backend project')
;

task('deploy:both', ['deploy:frontend', 'deploy:backend']);

...
```

but when I ran:
`dep deploy:both` or `dep deploy:frontend` or `dep deploy:backend`

`deployer` **always** asks me:
```
Select hosts: (comma separated)
  [0] frontend
  [1] backend
 >
```

And whichever host I select (also if I select both) I can see the execution of the deploy tasks, for example something like this
```
task deploy:setup
task deploy:lock
task deploy:release
task deploy:update_code
task deploy:shared
task deploy:writable
task deploy:symlink
task deploy:unlock
task deploy:cleanup
```

but I don't find anything on my remote host and I don't have any kind of error. Why? What am I doing wrong?

Thanks

**Comment by SiMoSiMo2** [comment]
hey @antonmedv, maybe you could help me

**Reply by antonmedv** [reply]
I can, but currently not have time.
